### DISCUSSION TITLE
On solvemp

### DISCUSSION THREAD

**Discussion by asxadf** [yalmip/YALMIP#1272] [Problems using YALMIP] [answered]
Hi Johan,

I am working with command solvemp to do the multiparametric programming. Yet, I am not sure whether this command still works. In my case, I slightly changed the model then the results were wired.

Could you please let me know whether solvemp still works with LP and MILP?

Thank you,
Chris

**Comment by johanlofberg** [comment]
Yes, it still works in the sense that my small tests still runs, I never use it actively so bugs could be rampant

You will have to first establish that it works for very very small problems (i.e. that you have a working installation) and then start complicating matters, because it could easily be that you are formulating a problem that just simply isn't possible to solve (too large, not in the supported problem class)

**Reply by asxadf** [reply]
```
clc;
clear;
close all;
%% ----------------------------- Decisions ----------------------------- %%
z = sdpvar(2, 1); %
x = sdpvar(2, 1); %
%
%% ----------------------------- Objective ----------------------------- %%
obj = 8.1*z(1) + 10.8*z(2);
%
%% ---------------------------- Constraints ---------------------------- %%
F = [];
F = F + [ 0.80*z(1) + 0.44*z(2) <= 24000 + x(1)          ];
F = F + [ 0.05*z(1) + 0.10*z(2) <=  2000          + x(2) ];
F = F + [ 0.10*z(1) + 0.36*z(2) <=  6000                 ];
F = F + [ z >= 0 ];
%
F = F + [ 0 <= x(1) <= 6000 ];
F = F + [ 0 <= x(2) <= 500 ];

% sol = optimize(F, obj);
% Ops = sdpsettings('solver', 'gurobi');
% optimize(F, obj, Ops);

%
[sol, diagnostics, aux, Valuefunction, OptimalSolution] = solvemp(F, -obj, [], x);

% assign(x, [0; 0]);
% [i,j] = isinside(sol{1}.Pn,x);
% sol{1}.Fi{j}*xx + sol{1}.Gi{j};

plot(Valuefunction);

```

**Reply by asxadf** [reply]
Hi Dr. Johan.

Thanks for reaching out!

I tried to use this small sample (maybe smaller than yours). I have solved it manually. So I wanted to try it on YALMIP+MPT. But it just failed. In the return of `sol`, the matrixes are empty and this really confuses me.

**Reply by johanlofberg** [reply, marked as answer]
Then your MPT is not working correctly

[sol, diagnostics, aux, Valuefunction, OptimalSolution] = solvemp(F, -obj, [], x);
mpt_plcp: 3 regions
-> Generated 1 partition.

Hence, you will have to put a break in callmpt3 and step through and see what happens when MPT is called on your machine

**Reply by asxadf** [reply]
Many thanks for the suggestions!
